mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/impl/MongoDBDatastoreProvider.java

```diff
@@ -168,4 +168,20 @@ private DB extractDatabase(MongoClient mongo, MongoDBConfiguration config) {
 			}
 		}
 	}
+
+	protected MongoDBConfiguration getConfig() {
+		return config;
+	}
+
+	protected MongoClient getMongoClient() {
+		return mongo;
+	}
+
+	protected void setMongoClient(MongoClient mongoClient) {
```

Comment:
I don't have any preference pattern-wise. Only the class should be "effectively immutable" after its initialization, i.e. no changes to its state should be done after `configure()` has been called. The setters now suggests that the state also can be altered later on. Thus I'd prefer to not have them.